### PR TITLE
TrackDAO: Reformat code / Avoid mutex locks on Track object

### PR DESCRIPTION
**Pull request by uklotzde**
This technical PR will reduce the number of merge conflicts that I already experience frequently.

- Reformat code in anonymous namespace, i.e. fix the indentation
- Bind database column upon insert/update from `TrackRecord` instead of `Track` to avoid repeated locking of mutexes. The additional copy shouldn't matter since all dynamically allocated C++ objects are implicitly shared (Qt)

Both changes affect a single part of the code.

### CODE REVIEW

File: src/library/dao/trackdao.cpp [hunk 1]
```diff
@@ -1870,10 +1914,10 @@ void TrackDAO::detectCoverArtForTracksWithoutCover(volatile const bool* pCancel,
     QSqlQuery updateQuery(m_database);
     updateQuery.prepare(
         "UPDATE library SET "
-        "  coverart_type=:coverart_type,"
-        "  coverart_source=:coverart_source,"
-        "  coverart_hash=:coverart_hash,"
-        "  coverart_location=:coverart_location "
+        "coverart_type=:coverart_type,"
```
Comment (daschuer) [inline src/library/dao/trackdao.cpp:1917]: Is a tailing " " lost here?

**Review by daschuer — COMMENTED** [review]
Just one comment, unsure. 

I wish we have a 2.3 branch, so that can go to master. 

@Holzhaus you working on the last blockers. Are you able to estimate something?

**Comment by Holzhaus** [comment]
> @Holzhaus you working on the last blockers. Are you able to estimate something?

AFAIC everything is ready to merge now.

File: src/library/dao/trackdao.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/dao/trackdao.cpp:1917]: Every line ends with either ' ' or with ',' consistently.

**Comment by daschuer** [comment]
OK, so lets wait here.

File: src/library/dao/trackdao.cpp (hunk 1 above)
Comment (daschuer) [reply src/library/dao/trackdao.cpp:1917]: Oh, now I see. Thanks

**Comment by uklotzde** [comment]
I have so many pending pull requests that I need to suspend working on Mixxx. This is becoming unmanageable.

**Comment by Be-ing** [comment]
I have just pushed the 2.3 branch so this can be merged to master now.

**Pull request merged**